src/lib/swapper/swappers/ThorchainSwapper/utils/getLatestThorTxStatusMessage.ts

```diff
@@ -0,0 +1,113 @@
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import prettyMilliseconds from 'pretty-ms'
+import { assertUnreachable } from 'lib/utils'
+
+import type { ThorNodeStatusResponseSuccess } from '../types'
+
+export const getLatestThorTxStatusMessage = (
+  response: ThorNodeStatusResponseSuccess,
+  hasOutboundTx: boolean,
+): { status: TxStatus; message: string | undefined } => {
+  const stages_reverse_ordered = [
+    'outbound_signed',
+    'outbound_delay',
+    'swap_finalised',
+    'swap_status',
+    'inbound_finalised',
+    'inbound_confirmation_counted',
+    'inbound_observed',
+  ] as const
+
+  for (const key of stages_reverse_ordered) {
+    switch (key) {
+      case 'inbound_observed': {
+        const obj = response.stages[key]
+        return {
+          message: obj.completed ? 'Inbound transaction observed' : 'Inbound transaction pending',
```

Comment:
it could be cool to try and translate this, but likely will have to be a follow-up because other swappers propagate text directly from an api call so it wont be straightforwards

Reply:
Maybe something in the realm of "transaction accepted/validated (?) by THOR, waiting on more info yadi yada"?

Reply:
sounds good, went with `Inbound transaction accepted by THOR`